src/item.cpp

```diff
@@ -2974,6 +2974,14 @@ units::volume item::base_volume() const
         return corpse_volume( corpse->size );
     }
 
+    if( count_by_charges() ) {
+        if( type->volume % type->stack_size == 0_ml ) {
+            return type->volume / type->stack_size;
+        } else {
+            return type->volume / type->stack_size + 1_ml;
+        }
+    }
+
```

Comment:
The catch here is we're dealing with quantities with units rather than just int/int.

> `(n+d-1)/d`

That formula was one of the first I reached for as well, but it turns out to be a bit messier than expected in practice, because `n` is `quantity<foo, unit>` whereas `d` is a scalar (and could in theory also be a quantity). 

We could obtain and manipulate the raw dimensionless value from the `quantity`, but having a modulus operator seems much cleaner approach.

Reply:
Ah, I feared that might be the case.  I guess the morally correct thing here is that we ought to have `divide_round_up_to_multiple_of(n, d, b)` where `b` is our base value we want the result to be a multiple of (in this case, `1_ml`).  That allows you to turn `n` into something dimensionless by dividing by `b`.  But I can see that might be getting more complicated than is warranted for this case.

Reply:
That opens up the possibility of `divide_round_up_to_multiple_of( 100_ml , 4 , 7_ml );` which a) would need even more steps to calculate, and b) doesn't really have any use cases. Implementing such a function "correctly" looks like more trouble than it's worth for very little benefit. Probably better to stop going down this rabbit hole and just leave it as is -- the modulus operator is mathematically sound for our purposes and will do just fine.

> clarify the intent

That, on the other hand, I can do something about. Will add doxygen comments and other docu to highlight `item::base_volume()` and `item::volume()` being rounded up, caution about how they should (not) be used for further arithmetic, etc. Should also put in warnings about `itype::volume` for good measure.